MazeHunt/src/Maze.java

```diff
@@ -0,0 +1,955 @@
+import java.awt.Color;
+import java.awt.Rectangle;
+import java.awt.event.KeyEvent;
+import java.util.ArrayList;
+import processing.core.PApplet;
+import processing.core.PImage;
+
+import java.awt.geom.Line2D;
+
+
+public class Maze extends Screen{
+
+	private DrawingSurface surface;
+	private Clues c = new Clues();
+	int z,m;
+	int a=300,b=35;
+	private Color exitRevel;
+	public int cluesFound=1;
+	public boolean clue1Found = false;
+	public boolean clue2Found = false;
+	public boolean clue3Found = false;
+	public boolean clue4Found = false;
+	private PImage rat;
+	private PImage cat;
+
+
+	private Clues clue;
+	private Rectangle player1;
+	private Rectangle player2;
+
+
+	private Rectangle tester;
+
+	/**
+	 * Constructs a maze 
+	 */
+	public Maze(DrawingSurface surface) {
+		super(345,255);
+		this.surface = surface;
+
+
+	}
+
+	
+	/**
+	 * Place to draw the maze and characters on
+	 * @param marker the place on with the maze and characters will be drawn on
+	 */
+	public void draw() {
+		int clues=0;
+		surface.pushStyle();
+		surface.background(176,224,230);   // Clear the screen with a white background
+		surface.stroke(0);     // Set line drawing color to white
+		surface.noFill();
+		surface.fill(0);
+		surface.textSize(5);
+		surface.text("Maze Hunt Reminder: \nMove the mouse around to run from the cat and find the clues. "
+				+ "Remember you can only leave \n the maze once you have found all the clues. Good luck!", 5, 15);
+		//		surface.background(176,224,230);
+
+		//Rectangle(x,y,angle,length)
+		//System.out.println("here");
+		// marker.background(255);
+
+		// circle.draw(marker);
+		// marker.rect(20, 10, 20, 10);
+
+		int x, y, w, h, gap;
+		x = 15;
+		y = 35;
+		w = 300;
+		h = 200;
+		gap = 30;
+
+		player1 = new Rectangle(z,m,19,19);
+
+		player2 = new Rectangle(a,b,17,17);
+
+
+		tester = new Rectangle(100,203,50,20);
+		//top entry
+		Line2D tope = new Line2D.Float(x, y, x+gap, y);//clue
+		//color
+		//tope.setStroke(new Color(176,224,230));
+		float tex1 = (float) tope.getX1();
+		float tey1 = (float) tope.getY1();
+		float tex2 = (float) tope.getX2();
+		float tey2 = (float) tope.getY2();
+		//surface.line(tex1, tey1, tex2, tey2);
+
+		//top line
+		Line2D top = new Line2D.Float(x+gap, y, x+w, y);
+		float tx1 = (float) top.getX1();
+		float ty1 = (float) top.getY1();
+		float tx2 = (float) top.getX2();
+		float ty2 = (float) top.getY2();
+		surface.line(tx1, ty1, tx2, ty2);
+
+		//right line
+		Line2D right = new Line2D.Float(x+w, y, x+w, y+h);
+		float rx1 = (float) right.getX1();
+		float ry1 = (float) right.getY1();
+		float rx2 = (float) right.getX2();
+		float ry2 = (float) right.getY2();
+		surface.line(rx1, ry1, rx2, ry2);
+
```

Comment:
This is a pretty intense class! It looks like you are trying really hard to avoid using arrays or ArrayLists, but that is very clearly the type of structure you should be using. Consider making this class **much much shorter** by creating an ArrayList of lines as a field, adding the appropriate lines to the list in the constructor (or some other "maze initialization" method), then using loops to both draw the lines and detect collisions on the lines.